File: LAB4/B.cpp
#include <bits/stdc++.h>
using namespace std;

struct Node{
    int data;
    Node* left;
    Node* right;

    Node(int x) : data(x), left(nullptr), right(nullptr){}
};

void bt_insert(Node*& root, int val){
    if(root == nullptr){
        root = new Node(val);
        return;
    }
    if(val < root->data) bt_insert(root->left, val);
    else bt_insert(root->right, val);
}

int subtree_size(Node* root) {
    if(root == nullptr) return 0;
    return 1 + subtree_size(root->left) + subtree_size(root->right);
}

Node* find(Node* root, int x) {
    if(root == nullptr) return nullptr;
    if(root->data == x) return root;
    if(x < root->data) return find(root->left, x);
    else return find(root->right, x);
}

int main(){

    int n, x;
    cin >> n;
    Node* root = nullptr;
    for(int i = 0; i<n; i++){
        cin >> x;
        bt_insert(root, x);
    }

    cin >> x;
    Node* nodeX = find(root, x);

    cout << subtree_size(nodeX);

    return 0;
}

File: LAB4/B.py
import sys


class Node:
    def __init__(self, val):
        self.data = val
        self.right = None
        self.left = None


def bt_insert(root, val):
    if root is None:
        return Node(val)
    if val < root.data:
        root.left = bt_insert(root.left, val)
    else:
        root.right = bt_insert(root.right, val)
    return root


def bt_size(root):
    if root is None:
        return 0
    cur = root
    return 1 + bt_size(root.left) + bt_size(root.right)


data = sys.stdin.read().split()

n = int(data[0])
k = int(data[-1])
nums = list(map(int, data[1 : 1 + n]))
root = None

for x in nums:
    root = bt_insert(root, x)

print(bt_size(root))
